### PR TITLE
MRG: fix issues raised by LGTM.com

### PR DESCRIPTION
**Pull request by agramfort**
see https://lgtm.com/projects/g/mne-tools/mne-python/alerts/?mode=list

there are more...

there are quite some false positive but it's pretty efficient still.

### CODE REVIEW

**Comment by larsoner** [comment]
I think that the analyses have started failing because I added `externals` to the skip list. Hopefully a merge will fix it. This is at least good enough incremental progress.

**Comment by larsoner** [comment]
This pull request **introduces 5 alerts** and **fixes 106** when merging fd41864892510ff7d517aeb0ffb044fd98e55367 into 63a8ce271eac9b2d488c353e04a3117aa1c5e772 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-c7a10015862deb137e72e9809c7af193c9593946)

**new alerts:**

* 4 for \_\_eq\_\_ not overridden when adding attributes
* 1 for Module is imported with &#39;import&#39; and &#39;import from&#39;

**fixed alerts:**

* 35 for Unused local variable
* 32 for Module is imported with &#39;import&#39; and &#39;import from&#39;
* 9 for Unused named argument in formatting call
* 7 for Encoding error
* 3 for Redundant assignment
* 2 for Module is imported more than once
* 2 for Syntax error
* 2 for Inconsistent equality and hashing
* 2 for Result of integer division may be truncated
* 2 for Unused exception object
* 2 for Property in old\-style class
* 1 for Wrong name for an argument in a call
* 1 for Unused import
* 1 for Raising a tuple
* 1 for List comprehension variable used in enclosing scope
* 1 for Unreachable code
* 1 for Suspicious unused loop iteration variable
* 1 for Variable defined multiple times
* 1 for Wrong number of arguments in a call

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: .lgtm.yml [hunk 1]
```diff
@@ -8,3 +8,6 @@ extraction:
     index:
       filters:
         - exclude: "**/*.js"
+queries:
+  - exclude: py/missing-equals
+  - exclude: py/import-and-import-from
```
Comment (agramfort) [inline .lgtm.yml]: maybe with this you can revert some annoying changes eg the "import nibabel as nib" that disappeared and stuff related to os vs. op

I don't like that we don't keep the origin of the function in the call :)

Comment (larsoner) [reply .lgtm.yml]: :(

**Comment by agramfort** [comment]
want me to do it?

>

**Comment by larsoner** [comment]
No I did it and it wasn't as annoying as I thought it would be :)

**Comment by massich** [comment]
ok I'm on the party too. How can I help :)

**Comment by agramfort** [comment]
LGTM ! :)

**Comment by larsoner** [comment]
@massich this one is done, #5900 is more interesting and needs merge

**Comment by larsoner** [comment]
This pull request **introduces 6 alerts** and **fixes 106** when merging 28bfc346ce7018ddfbe6fbb92414cd30c7f08fbd into 6fdfb41246a1ca1a5ae2c381c5f7d97137a63265 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-e952ae3a3026732e71f0f44e7ec7a30c0f40e918)

**new alerts:**

* 4 for \_\_eq\_\_ not overridden when adding attributes
* 2 for Module is imported with &#39;import&#39; and &#39;import from&#39;

**fixed alerts:**

* 35 for Unused local variable
* 32 for Module is imported with &#39;import&#39; and &#39;import from&#39;
* 9 for Unused named argument in formatting call
* 7 for Encoding error
* 3 for Redundant assignment
* 2 for Module is imported more than once
* 2 for Syntax error
* 2 for Inconsistent equality and hashing
* 2 for Result of integer division may be truncated
* 2 for Unused exception object
* 2 for Property in old\-style class
* 1 for Wrong name for an argument in a call
* 1 for Unused import
* 1 for Raising a tuple
* 1 for List comprehension variable used in enclosing scope
* 1 for Unreachable code
* 1 for Suspicious unused loop iteration variable
* 1 for Variable defined multiple times
* 1 for Wrong number of arguments in a call

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by larsoner** [comment]
LGTM seems to have ignored our ignores, but we can live with the 6 warnings for now. I'm hoping merging will fix their parsing bug.

**Comment by agramfort** [comment]
Thx

**Comment by massich** [comment]
great!! now we can start bugging people in making LGTM green :)

**Pull request merged**